tests/system/Database/Live/InsertTest.php

```diff
@@ -79,13 +80,41 @@ public function testInsertBatch(): void
             ],
         ];
 
-        $this->db->table($table)->insertBatch($data);
+        $count = $this->db->table($table)->insertBatch($data);
+
+        $this->assertSame(2, $count);
 
         $expected = $data;
         $this->seeInDatabase($table, $expected[0]);
         $this->seeInDatabase($table, $expected[1]);
     }
 
+    public function testInsertBatchFailed(): void
+    {
+        $this->expectException(DatabaseException::class);
+
+        $data = [
+            [
+                'name' => 'Grocery Sales',
+            ],
+            [
+                'name' => null,
+            ],
+        ];
+
+        $db = $this->db;
+
+        if ($this->db->DBDriver === 'MySQLi') {
+            // strict mode is required for MySQLi to throw an exception here
+            $config                    = config('Database');
```

Comment:
can you not use the shared config so the strict mode will not persist in other tests?

Reply:
I thought the factories were reset during `setUp()`?

https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Test/CIUnitTestCase.php#L58

Reply:
Oh, ok forgot that.